shade_janitor/cleanup.py

```diff
@@ -1,65 +1,152 @@
 #!/usr/bin/env python
 
 
+def cleanup_instances(cloud, instances):
+    """Cleanup instances."""
+    for uuid in instances:
+        cloud.delete_server(uuid, wait=True, delete_ips=True)
+
+
+def dry_cleanup_instances(instances):
+    """Dry cleanup of instances."""
+    for uuid in instances:
+        print('nova delete {}'.format(uuid))
+
+
+def dry_cleanup_router_interfaces(routers_id, subnets_id):
+    """Dry cleanup of router interfaces."""
+    for r_uuid in routers_id:
+        for uuid in subnets_id:
+            print('neutron router-interface-delete {0} {1}'.format(
+                r_uuid, uuid))
+
+
+def cleanup_router_interfaces(cloud, routers_id, subnets_id):
+    """Cleanup router interfaces."""
+    for r_uuid in routers_id:
+        for uuid in subnets_id:
+            for router in cloud.search_routers(r_uuid):
+                cloud.remove_router_interface(router, uuid)
+
+
+def setup_router_interfaces_cleanup(routers, router_interfaces, dry_run):
+    """Setup router interfaces cleanup."""
+    routers_id = []
+    subnets_id = []
+    for r_uuid in routers:
+        routers_id.append(r_uuid)
+    for uuid in router_interfaces:
+        subs = router_interfaces[uuid]['subnet_ids']
+        subnets_id.extend(subs)
+    return routers_id, subnets_id
```

Comment:
If you are going to return here then we don't need the if/else below otherwise we don't want this return here.  
I'm thinking it's the later since you can have multiple subnets/routers that might need cleaned up a single return isn't that useful.